assets/src/edit-story/app/rightClickMenu/useRightClickMenu.js

```diff
@@ -0,0 +1,173 @@
+ * - A ref for the container where the right click menu will be rendered
+ */
+const useRightClickMenu = () => {
+  const { selectedElements } = useStory(
+    ({
+      state: { currentPage, selectedElements, selectedElementAnimations },
+    }) => ({
+      currentPage,
+      selectedElements,
+      selectedElementAnimations,
+    })
+  );
+
+  // Ref to use to calculate the menu's position
+  const rightClickAreaRef = useRef();
+
+  const [isMenuOpen, setIsMenuOpen] = useState(false);
+  const [menuPosition, setMenuPosition] = useState({ x: 0, y: 0 });
+
+  /**
+   * Open the menu at the position from the click event.
+   */
+  const handleOpenMenu = useCallback((evt) => {
+    evt.preventDefault();
+    const layoutRect = rightClickAreaRef?.current?.getBoundingClientRect();
+
+    if (layoutRect) {
+      setMenuPosition({
+        x: evt.clientX - layoutRect?.left,
+        y: evt.clientY - layoutRect?.top,
+      });
+
+      setIsMenuOpen(true);
+    }
+  }, []);
+
+  /**
+   * Close the menu and reset the tracked position.
+   */
+  const handleCloseMenu = useCallback(() => {
+    if (isMenuOpen) {
+      setIsMenuOpen(false);
+      setMenuPosition({ x: 0, y: 0 });
+    }
+  }, [isMenuOpen]);
+
+  /**
+   * Prevent right click menu from removing focus from the canvas.
+   */
+  const handleMouseDown = useCallback((ev) => {
+    ev.stopPropagation();
+  }, []);
+
+  const selectedElement = selectedElements?.[0];
+
+  const menuItemProps = useMemo(
+    () => ({
+      handleMouseDown,
+    }),
+    [handleMouseDown]
+  );
+
+  const defaultItems = useMemo(
+    () => [
+      {
+        label: RIGHT_CLICK_MENU_LABELS.COPY,
+        shortcut: '⌘ X',
+        onClick: noop,
+        ...menuItemProps,
+      },
+      {
+        label: RIGHT_CLICK_MENU_LABELS.PASTE,
+        shortcut: '⌘ V',
+        onClick: noop,
+        ...menuItemProps,
+      },
+      {
+        label: RIGHT_CLICK_MENU_LABELS.DELETE,
+        shortcut: 'DEL',
+        onClick: noop,
+        ...menuItemProps,
+      },
+    ],
+    [menuItemProps]
+  );
+
+  const pageItems = useMemo(
+    () => [
+      ...defaultItems,
+      {
+        label: RIGHT_CLICK_MENU_LABELS.DUPLICATE_PAGE,
+        onClick: noop,
+        ...menuItemProps,
+      },
+      {
+        label: RIGHT_CLICK_MENU_LABELS.DELETE_PAGE,
+        onClick: noop,
+        ...menuItemProps,
+      },
+    ],
+    [defaultItems, menuItemProps]
+  );
+
+  const menuItems = useMemo(() => {
+    switch (selectedElement?.type) {
+      case ELEMENT_TYPE.IMAGE:
+      case ELEMENT_TYPE.SHAPE:
+      case ELEMENT_TYPE.TEXT:
+      case ELEMENT_TYPE.VIDEO:
+      default:
+        return pageItems;
+    }
+  }, [pageItems, selectedElement?.type]);
+
+  // Override the browser's context menu
+  useEffect(() => {
+    document.addEventListener('contextmenu', handleOpenMenu);
+
+    return () => {
+      document.removeEventListener('contextmenu', handleOpenMenu);
+    };
+  });
```

Comment:
I wonder if this is better just to attach to the canvas layout area so this functionality is more contained:
<img width="1077" alt="Screen Shot 2021-06-11 at 12 01 17 PM" src="https://user-images.githubusercontent.com/35983235/121724104-02cbdd80-caad-11eb-922d-578a92adcade.png">

Reply:
Yeah, dope idea!

Reply:
I attached the ref to the `PageAreaContainer` which is the canvas container (Unless I am mistaken)